### ISSUE TITLE
Round-trip problem with empty params

### ISSUE THREAD

**Issue by daniel-rikowski** [activerecord-hackery/ransack#366]
**tl;dr**: I've noticed a small problem with *empty* search queries if the params are piped through `url_for` (or `link_to`) and then redirected to Ransack again.

**Long example:**

1. This url is sent to Rails: `/orders?utf8=✓&q[id_eq]=&q[order_code_cont]` and subsequentially processed by Ransack via `Order.search(params[:q])`.
2. **Before** `Order.search` is executed the params hash looks like this:

```ruby
{
  'action' => 'index'
  'controller' => 'orders'
  'q' => {
    'id_eq' => ''
    'order_code_cont' => ''
  }
}
```

3\. **After** `Order.search` is executed the params hash looks like this:
```ruby
{
  'action' => 'index'
  'controller' => 'orders'
  'q' => {
  }
}
```
Most likely this is because of #201 

4\. The actual problem shows when I'm using `url_for` (in this case to add a `page` param)
`url_for(params.merge(page: 2))` generates this url: `/orders?utf8=✓&q=&page=1`

5\. Suddenly Ransack has to handle an (empty) string instead of a hash which leads to various exceptions.

I can think of two ways to fix this: Either don't change `params` (e.g. use `params.dup` before calling `delete_if`) or use `params.presence` at the necessary locations. (At `Ransack::Adapters::ActiveRecord::Base.ransack` and `Ransack::Helpers::FormHelper.sort_link`)

Thank you in advance!

**Comment by jonatack** [comment]
Fixed with 28433f9.
